dartsim/src/CustomHeightmapShape.cc

```diff
@@ -33,37 +34,52 @@ CustomHeightmapShape::CustomHeightmapShape(
     int _subSampling)
   : dart::dynamics::HeightmapShape<float>()
 {
-  double heightmapSizeZ = _input.MaxElevation();
+  float heightmapSizeZ = _input.MaxElevation();
   const bool flipY = false;
   const int vertSize = (_input.Width() * _subSampling) - _subSampling + 1;
+
   math::Vector3d scale;
   scale.X(_size(0) / vertSize);
   scale.Y(_size(1) / vertSize);
 
-  if (math::equal(heightmapSizeZ, 0.0))
+  if (math::equal(heightmapSizeZ, 0.0f))
     scale.Z(1.0);
   else
     scale.Z(fabs(_size(2)) / heightmapSizeZ);
 
   auto sizeIgn = ignition::math::eigen3::convert(_size);
 
-  // We need to make a copy here in order to use the non-const FillHeightMap
+  std::vector<float> heightsFloat;
+
+  // We need to make a copy below in order to use the non-const FillHeightMap
```

Comment:
I just noticed this complaint, so I made the `FillHeightmap` function `const` in https://github.com/ignitionrobotics/ign-common/pull/292/commits/8c63dac5e9ecc7495e33ca65657b6ed12d7ece18 in https://github.com/ignitionrobotics/ign-common/pull/292

I think we can skip copying the data with this change

Reply:
Thanks! d78cd486d264ebe122e58566a7ef243316f5dba6